From ac5430e2ea3f60b930ce29a2f95230058a7b81bd Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Wed, 18 Oct 2017 20:11:48 +0200
Subject: [PATCH 21/80] SettingsLib: Add LineageParts settings to tile list

* By adding our own package we can inject settings which also respect
  ordering by priority
* This way we don't have to modify xml files manually when using IA

Change-Id: Iadf05092e8b8512dc07f9fa2305c36fde10084c1
---
 .../android/settingslib/drawer/TileUtils.java    | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/packages/SettingsLib/src/com/android/settingslib/drawer/TileUtils.java b/packages/SettingsLib/src/com/android/settingslib/drawer/TileUtils.java
index 0f0e4e57886..7f50b3aa18b 100644
--- a/packages/SettingsLib/src/com/android/settingslib/drawer/TileUtils.java
+++ b/packages/SettingsLib/src/com/android/settingslib/drawer/TileUtils.java
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2015 The Android Open Source Project
+ * Coypright (C) 2017 The LineageOS Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -93,6 +94,9 @@ public class TileUtils {
     private static final String MANUFACTURER_DEFAULT_CATEGORY =
             "com.android.settings.category.device";
 
+    private static final String LINEAGE_SETTINGS_ACTION =
+            "org.lineageos.lineageparts.action.SETTINGS";
+
     /**
      * The key used to get the category from metadata of activities of action
      * {@link #EXTRA_SETTINGS_ACTION}
@@ -184,6 +188,8 @@ public class TileUtils {
 
     public static final String SETTING_PKG = "com.android.settings";
 
+    public static final String LINEAGE_SETTING_PKG = "org.lineageos.lineageparts";
+
     /**
      * Build a list of DashboardCategory. Each category must be defined in manifest.
      * eg: .Settings$DeviceSettings
@@ -226,6 +232,10 @@ public class TileUtils {
                 // Only add Settings for this user.
                 getTilesForAction(context, user, SETTINGS_ACTION, cache, null, tiles, true,
                         settingPkg);
+                if (SETTING_PKG.equals(settingPkg)) {
+                    getTilesForAction(context, user, LINEAGE_SETTINGS_ACTION, cache, null, tiles,
+                        true, LINEAGE_SETTING_PKG);
+                }
                 getTilesForAction(context, user, OPERATOR_SETTINGS, cache,
                         OPERATOR_DEFAULT_CATEGORY, tiles, false, true, settingPkg);
                 getTilesForAction(context, user, MANUFACTURER_SETTINGS, cache,
@@ -294,8 +304,10 @@ public class TileUtils {
                 continue;
             }
             category.title = resolved.activityInfo.loadLabel(pm);
-            category.priority = SETTING_PKG.equals(
-                    resolved.activityInfo.applicationInfo.packageName) ? resolved.priority : 0;
+            String pkgName = resolved.activityInfo.applicationInfo.packageName;
+            category.priority = (SETTING_PKG.equals(pkgName) || LINEAGE_SETTING_PKG.equals(pkgName))
+                    ? resolved.priority
+                    : 0;
             if (DEBUG) Log.d(LOG_TAG, "Adding category " + category.title);
         }
 
-- 
2.17.1

